### DISCUSSION TITLE
"Resource Exhausted" error while running lmp

### DISCUSSION THREAD

**Discussion by jaejae9804** [deepmodeling/deepmd-kit#3962] [Q&A] [answered]
Hi everyone,
greatly appreciate in advance for everyone. What I was doing is running LAMMPS with a Cu dp model developed earlier and encoutered an error metioning what seems to be related to the memory usage. Ive attached the error log, with the input scripts Ive used. 
Im using Geforce rtx 4090 as my local computer, and the environment Im using to run deepmd-kit was created with nvidia-docker.
To be specific, the environment Im using to run deepmd-kit is what follows:

Ubuntu-22.04.3 LTS
Geforce rtx 4090 
nvidia-docker
docker image: deepmodeling/deepmd-kit:2.2.7_cuda11.6_gpu
(docker container run -it -v /home/jaejae:/root --name dp_2.7 --gpus all deepmodeling/deepmd-kit:2.2.7_cuda11.6_gpu)

If checking out the structural file attached below, you can see that Ive used 100,0188 Cu atoms for LAMMPS testing. When running the same simulation with the identical input scripts with 4000atoms, everything seems to be fine, meaning that none of the following errors happened. It would be of a great help if anyone can give an advice on how to solve this problem. Thanks.
[error log.txt](https://github.com/user-attachments/files/16154919/error.log.txt)
[in.lammps.txt](https://github.com/user-attachments/files/16154920/in.lammps.txt)
[xo.lmp.txt](https://github.com/user-attachments/files/16154939/xo.lmp.txt)

**Comment by njzjz** [comment]
You can use the MPI to run on multiple GPU cards if one card doesn't have enough memory.

**Reply by jaejae9804** [reply]
Thank you for the response. Unfortunately, I do have only one gpu card in my local computer...If in my case, does mpirun have benefit in fixing this problem?

**Reply by jaejae9804** [reply]
Oh, just in case Ive been less specific, the memory of the gpu card  is 32GB, and the additional infos about the cpus are as follows:

jaejae@DESKTOP-ORMR4QC:~$ free
               total        used        free      shared  buff/cache   available
Mem:        32707244     2529324    28396780      347484     1781140    29428268
Swap:        8388608           0     8388608


jaejae@DESKTOP-ORMR4QC:~/Sn/dpgen/data/b763195f5224117522baa37fbf29673885f91154/003$ lscpu
Architecture:            x86_64
  CPU op-mode(s):        32-bit, 64-bit
  Address sizes:         46 bits physical, 48 bits virtual
  Byte Order:            Little Endian
CPU(s):                  28
  On-line CPU(s) list:   0-27
Vendor ID:               GenuineIntel
  Model name:            Intel(R) Core(TM) i7-14700K
    CPU family:          6
    Model:               183
    Thread(s) per core:  2
    Core(s) per socket:  14
    Socket(s):           1
    Stepping:            1

**Reply by njzjz** [reply, marked as answer]
To run 1 million atoms with a compressed model, about 60GB of GPU memory is required. Please see Table 5 at https://doi.org/10.1063/5.0155600.
